src/linear_algebra/basis.lean

```diff
@@ -1424,3 +1424,52 @@ let ⟨q, hq⟩ := p.exists_is_compl in nonempty.intro $
   (prod_equiv_of_is_compl q p hq.symm)
 
 end division_ring
+
+section restrict_scalars
+
+variables [comm_ring R] [ring K] [nontrivial K] [add_comm_group V]
+variables [algebra R K] [module K V] [module R V]
+variables [is_scalar_tower R K V] [no_zero_smul_divisors R K] (b : basis ι K V)
+variables (R)
+
+open submodule
+
+/-- Let `b` be a `K`-basis of `V`. Let `R` be a comm_ring such that `algebra R K` with no zero
+smul divisors, then the submodule of `V` spanned by `b` over `R` admits `b` as a `R`-basis. -/
+noncomputable def basis.restrict_scalars : basis ι R (span R (set.range b)) :=
```

Comment:
I was wondering if this needs to be connected to [`basis.localization_localization`](https://leanprover-community.github.io/mathlib_docs/ring_theory/localization/module.html#basis.localization_localization) but that seems different enough that this way is fine.

Reply:
Are they similar enough that a docstring connection is worth having?